src/main/java/teammates/sqllogic/core/DataBundleLogic.java

```diff
@@ -485,8 +485,10 @@ private static void linkEntities(SqlDataBundle dataBundle) {
             usersMap.put(student.getId(), student);
             Course course = coursesMap.get(student.getCourse().getId());
             student.setCourse(course);
```

Comment:
@DhiraPT Actually what is the point of this? 

If the student object itself already has a course object that is not null, you retrieve the corresponding course again based on the ID that the student has and set it again?